tcn_hpl/data/tcn_dataset.py

```diff
@@ -0,0 +1,139 @@
+import torch
+from torch.utils.data import Dataset
+import kwcoco
+from tqdm import tqdm
+import logging
+import tcn_hpl.utils.utils as utils
+
+logger = logging.getLogger()
+logging.basicConfig(level=logging.INFO)
+
+
+def standarize_online_inputs(inputs):
+    pass
+
+def standarize_offline_inputs(data):
+    dset = data[0]
+    frames = data[1]
+
+
+def collect_inputs(data, offline=True):
+    """
+    Collects inputs from either an offline or real-time source.
+
+    Args:
+        data: data stream input. For offline, a coco dataset. For online, a set of ROS messages
+            
+        offline (bool): If True, fetches data from offline source; otherwise, real-time.
+
+    Returns:
+        inputs: The collected inputs from either a real-time or offline source in standarized format.
+            Designed as a dict of lists. all lists must have the size of the widnow_size:
+                {object_dets: [], pose_estimations:[], left_hand: [], right_hand: []}
+                
+    """
+    
+    
+    if offline:
+        inputs = standarize_offline_inputs(data)
+    else:
+        inputs = standarize_online_inputs(data)
+    
+    
+    return inputs
+
+def define_tcn_vector(inputs):
+    """
+    Define the TCN vector using the collected inputs.
+
+    Args:
+        inputs: The inputs collected from either real-time or offline source.
+
+    Returns:
+        tcn_vector: The defined TCN vector.
+    """
+    tcn_vector = torch.tensor(inputs).float()  # Dummy transformation
+    return tcn_vector
+
+
+class TCNDataset(Dataset):
+    def __init__(self, kwcoco_path: str, sample_rate: int, window_size: int):
```

Comment:
I was thinking that this dataset could take three paths as input: for object detection COCO, for pose COCO, and for activity truth COCO. If this is one combined file, we will still have to create this file outside of usage of this dataset whenever we change our detection or pose sources. I was thinking that if they are separate inputs, we could more easily configure mixing and matching for training experiments. Thoughts?

Reply:
The current offline pipeline works sequentially: we generate a detection kwcoco and input it to the pose generation script to populate it further with pose annotations. The script I wrote above takes the detection+pose combined kwcoco an populate it with activity GT. This would mean that we need to change the way we store our kwcoco as well, and regenerate them before continuing. 

For offline training there shouldn't be a change of data sources or kwcoco generation pipeline. Outside offline training , I currently set up a "collect_input" function outside the dataset class, which should take care of data standardization for either offline or online data inputs.

Reply:
it is not off the mark, though it depends on how the current code will be changed. If you want to simply take the new pose/detection models and replace them in the current code, then the pipeline stays the same (i.e. sequential input-output of kwcoco files and obtaining a unified kwcoco). It is probably possible to just create a new kwcoco file for each task (instead of adding annotations) easily enough, but we now generate more kwcoco files without improving or adding functionality. 

The way I see it, there isnt anything inherently different in creating 3 kwcoco files versus 1 kwcoco file, since this is only being used in the offline system (since the online system gets its own logic written in the "collect_inputs" function). Also - with 3 kwcoco files, you will need to cross reference pose, detections, and activity GT to the same images, and will also need to ensure that frames have the same ids in all those kwcocos.